src/utils/embeddingPlotHelperFunctions/helpers.js

```diff
@@ -3,12 +3,22 @@ import * as d3 from 'd3-scale';
 
 import { union } from '../cellSetOperations';
 
+const colorInterpolator = d3Chromatic.interpolatePurples;
+
 const hexToRgb = (hex) => {
   if (hex) {
-    return hex.replace(/^#?([a-f\d])([a-f\d])([a-f\d])$/i,
-      (m, r, g, b) => `#${r}${r}${g}${g}${b}${b}`)
-      .substring(1).match(/.{2}/g)
-      .map((x) => parseInt(x, 16));
```

Comment:
I think this was me copy-pasting some code from stackoverflow :D Nice refactoring, much much cleaner